### PR TITLE
Add more options to jenkins.yml for deployments

### PR DESCRIPTION
**Pull request by pmac**
* Add ability to toggle smoke tests
* Add ability to toggle deployment
* Add ability to toggle integration tests
* Add ability to toggle pushing docker images to public registry
* Add branch name to deploy image
* Add bash utility for setting git environment variables

### CODE REVIEW

**Comment by pmac** [comment]
This addresses most of #4682.

File: docker/bin/push2privateregistries.sh [hunk 1]
```diff
@@ -7,12 +7,14 @@
...
 
+DOCKER_TAG="${BRANCH_NAME}-${GIT_COMMIT}"
+
 # Push to private registry
 for PRIVATE_REGISTRY in ${PRIVATE_REGISTRIES//,/ };
 do
     for DEIS_APP in ${DEIS_APPS//,/ };
     do
-        docker tag $FROM_DOCKER_REPOSITORY:${GIT_COMMIT} $PRIVATE_REGISTRY/$DEIS_APP:${GIT_COMMIT}
+        docker tag $FROM_DOCKER_REPOSITORY:${DOCKER_TAG} $PRIVATE_REGISTRY/$DEIS_APP:${GIT_COMMIT}
```
Comment (pmac) [inline docker/bin/push2privateregistries.sh]: Don't need to use the branch-name tag in the private registry since it's already tagged per deis app name.

**Comment by pmac** [comment]
I'm saving the bit about allowing deployment of demo-like apps via the `branches` config area of `jenkins.yml` for later as it's a much larger change. I'm thinking we'll eventually have a single pipeline flow that will work out whether to do a demo-style or prod-style deployment on the fly either via branch name or via `demo: true` in the YAML. I think that part is for after a docker development environment is done.

**Comment by pmac** [comment]
Oh, and I pushed it to the `run-integration-tests` branch and the [test results look good](https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/run-integration-tests/8/pipeline).

**Comment by pmac** [comment]
@jgmize I'm not actually sure if assigning to you sends you email, so I'm pinging you manually as well for an r?

@glogiotatidis I'd appreciate it if you had a look as well if you have time.

**Comment by pmac** [comment]
@jgmize you said in a meeting that containers are being left running on Jenkins. Turns out I broke cleanup when I changed `local` to `smoke` for the name of the functional tests. The last commit in this PR fixes that. Perhaps we can bump up review and merge of this in the queue?

**Comment by jgmize** [comment]
@pmac and I discussed the edge case I ran into in https://ci.us-west.moz.works/blue/organizations/jenkins/bedrock_multibranch_pipeline/detail/demo__pr4684/1/pipeline/ and adding an `--rm` to the `docker run` commands should eliminate the name conflict.

**Comment by pmac** [comment]
@jgmize I've added said changes in a new commit. Thanks for the consult!
